src/main/java/com/linecorp/armeria/server/metrics/MetricsServiceCodec.java

```diff
@@ -0,0 +1,71 @@
+package com.linecorp.armeria.server.metrics;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.Timer;
+
+import com.linecorp.armeria.common.ServiceInvocationContext;
+import com.linecorp.armeria.common.SessionProtocol;
+import com.linecorp.armeria.common.metrics.RequestMetrics;
+import com.linecorp.armeria.server.DecoratingServiceCodec;
+import com.linecorp.armeria.server.ServiceCodec;
+import com.linecorp.armeria.server.ServiceConfig;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.channel.Channel;
+import io.netty.util.concurrent.Promise;
+
+class MetricsServiceCodec extends DecoratingServiceCodec {
```

Comment:
naming: class name is quite indistinguishable with `MetricCollectingServiceCodec`, how about prefix(or suffix) `Dropwizard`?